src/mon/FSCommands.cc

```diff
@@ -681,6 +681,38 @@ class SetHandler : public FileSystemCommandHandler
         fs->mds_map.set_min_compat_client(vno);
       };
       fsmap.modify_filesystem(fs->fscid, std::move(f));
+    } else if (var == "reject_client_session") {
+      bool reject_session = false;
+      int r = parse_bool(val, &reject_session, ss);
+      if (r != 0) {
+        return r;
+      }
+
+      if (reject_session) {
+        if (!(fs->mds_map.test_flag(CEPH_MDSMAP_REJECT_CLIENT_SESSION))) {
+          fsmap.modify_filesystem(
+            fs->fscid,
+            [n](std::shared_ptr<Filesystem> fs)
+          {
+            fs->mds_map.set_flag(CEPH_MDSMAP_REJECT_CLIENT_SESSION);
+          });
+          ss << "client blocked from establishing new sessions"; 
+        } else {
+          ss << "client already blocked from establishing new sessions";
+        }     
+      } else {
+          if (fs->mds_map.test_flag(CEPH_MDSMAP_REJECT_CLIENT_SESSION)) {
+            fsmap.modify_filesystem(
+              fs->fscid,
+              [n](std::shared_ptr<Filesystem> fs)
+            {
+              fs->mds_map.clear_flag(CEPH_MDSMAP_REJECT_CLIENT_SESSION);
+            });
+            ss << "client unblocked from establishing new sessions"; 
+          } else {
+            ss << "client already unblocked from establishing new sessions";
```

Comment:
"clients allowed to establish new sessions" should be a better message

also, do we really need to differentiate between the two cases in either of the scenarios
i.e. current state and requested state of the flag

Reply:
@mchangir it's just that I don't want to allow unnecessary modification to fsmap and to the flag. So first it checks if its set/unset.

Reply:
> "clients allowed to establish new sessions" should be a better message

Cool, I'll make that change.

Reply:
@mchangir modified the message. thanks.